dynamiqs/sparse.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import equinox as eqx
+import jax.numpy as jnp
+from jaxtyping import Array, ArrayLike, Scalar, ScalarLike
+
+from .qarray import QArray
+
+
+class SparseQArray(QArray):
+    diags: Array
```

Comment:
I will have a go at it, I don't think it would be that harmful, we already iterate through the diagonals and slice them so your suggestion makes sense.

Reply:
Then we could also store `diags` and `offsets` as a dict `{offset: diag}`.